### DISCUSSION TITLE
combining --list-tests and --filter based on fully qualified name?

### DISCUSSION THREAD

**Discussion by jeroenjanssen-cpp** [microsoft/testfx#4111] [Q&A] [answered]
Hello,

Is there a way to get the fully qualified name of a test from the commandline with --list-tests? (we are experimenting with the new Testing.Platform)

The --list-tests does not show the fully qualified name of a test
The --filter option is able to run tests based on the filter and it accepts the fully qualified name

We have some testcases with overlapping names (e.g. A and A_B, A_C, A_D, B, B_A, B_B)
With --list-tests we get A, which when passed in a --filter will include running A_B, A_C, A_D instead of only A.

I would expect this can be solved if we could get the fully qualified name of a test

The reason behind all this, is that we are investigating some unstable tests and would like to run the tests in real isolation for analysis, which is why we have a script that discovers all the testcases and then runs them separately, one by one (possibly with different arguments)

**Comment by Evangelink** [comment, marked as answer]
It's currently not possible as we have mainly copied VSTest behavior on that front but we have opened this ticket #3221 as I wanted to experiment with building more complex scenario out of display list of tests.
